### PR TITLE
[server] allow having +1 active tokens

### LINKED ISSUE #7196: Bitbucket integration - Permissions expire at least once per day automatically
### Bug description

At least once per day, permissions granted to Gitpod for Bitbucket integration get reset for no apparent reason.

Usually, I become aware of this because of an auth error I get when I try to push my work.

So, I have to disconnect and reconnect Bitbucket from the Integration dashboard. This step is pretty fast, because credentials are not required again and no new tokens are issued by Bitbucket.

### Steps to reproduce

1. Connect Bitbucket
2. Create or Open any Workspace
3. Wait at most 24 hours
4. Try to push any commit

### Workspace affected

cyan-echidna-eskb066m.ws-eu23.gitpod.io

### Expected behavior

_No response_

### Example repository

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by AlexTugarev**
We've seen cases where more than one active token is persisted in the DB, which might be cased by db-sync issues. Given that this case is quite rare, this PR drops the hard assertion and replaces it by logging a warning. This is a mitigation for #7196.

Fixes #7196

```release-note
Fix "token not found" issues.
```

### CODE REVIEW

**Comment by roboquat** [comment]
@AlexTugarev: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **alextugarev** after the PR has been reviewed.

Associated issue: *#7196*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by geropl — APPROVED** [review]
Thx @AlexTugarev for picking this up again, I nearly forgot! :+1:
Code LGTM

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7837#pullrequestreview-863519546" title="Approved">geropl</a>*

Associated issue: *#7196*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
